modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/notification-sidebar/index.tsx

```diff
@@ -0,0 +1,202 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import classNames from 'classnames';
+import {useEffect, useState} from 'react';
+
+import './index.scss';
+import {getApplicationsById} from '../../services/Application';
+import {
+	getUserNotification,
+	putUserNotificationRead,
+} from '../../services/notification';
+import {getLiferaySiteName} from '../../utils/liferay';
+import {PostType} from './postTypes';
+
+const initialPagination = {order: 'desc', page: 1, pageSize: 2, totalCount: 0};
+
+const NotificationSidebar: React.FC = () => {
+	const [posts, setPosts] = useState<PostType[]>([]);
+	const [totalCount, setTotalCount] = useState<number>(
+		initialPagination.totalCount
+	);
+	const [postsWithLinks, setArrayOfLinks] = useState<PostType[]>([]);
+	const [page, setPage] = useState<number>(initialPagination.page);
+	const hasMorePostsToLoad = posts.length < totalCount;
+
+	const notificationCategory = {
+		Application: 'Application ',
+	};
+
+	const creatRoute = (externalReferenceCode: string, entity: string) => {
+		const currentSiteName = getLiferaySiteName();
+		const link = `${currentSiteName}/${entity}?externalReferenceCode=${externalReferenceCode}`;
+
+		return link;
+	};
+
+	const markAsRead = (post: PostType) => {
+		if (!post.read) {
+			putUserNotificationRead(post.id);
+		}
+	};
+
+	const extractNumber = (message: string | undefined) => {
+		const number = message?.match(/\d/g);
+
+		return Number(number?.join(''));
+	};
+
+	async function getNotifications() {
+		try {
+			const response = await getUserNotification(
+				initialPagination.pageSize,
+				page,
+				initialPagination.order
+			);
+			const notifications = response?.data;
+
+			if (notifications) {
+				setTotalCount(notifications.totalCount);
+				setPosts((previousPosts: PostType[]) => [
+					...previousPosts,
+					...notifications.items,
+				]);
+			}
+
+			return response;
+		}
+		catch (error) {
+			console.error('Error getting notifications:', error);
+			throw error;
+		}
+	}
+
+	async function getExternalReferenceCode(id: number) {
+		try {
+			const response = await getApplicationsById(
+				id,
+				'externalReferenceCode'
+			);
+			const data = response?.data?.items?.[0]?.externalReferenceCode;
+
+			if (!data) {
+				throw new Error('External reference code not found');
+			}
+
+			return data;
+		}
+		catch (error) {
+			console.error(
+				`Error fetching external reference code for ID ${id}: ${error}`
+			);
+			throw error;
+		}
+	}
+
+	const generateLinks = async () => {
+		const newLinks = await Promise.all(
+			posts.map(async (post) => {
+				const postId = extractNumber(post.message);
+				const isMatchingApplication = post.message?.includes(
+					notificationCategory.Application + postId
+				);
+
+				const genericRoute = '#!';
+				if (isMatchingApplication) {
```

Comment:
Please add an empty line here.
```suggestion

				if (isMatchingApplication) {
```